test/src/redirect_test.coffee

```diff
@@ -48,3 +48,10 @@ describe 'XMLHttpRequest', ->
         expect(headers).not.to.have.property 'content-length'
         done()
       @xhr.send 'This should be dropped during the redirect'
+
+    it 'provides the final responseURL', (done) ->
```

Comment:
This is the case that I was worried about. Thank you very much for the thorough tests!